homeassistant/components/stream/core.py

```diff
@@ -137,17 +215,35 @@ def get_segments(self) -> deque[Segment]:
         self._idle_timer.awake()
         return self._segments
 
-    async def recv(self) -> Segment | None:
+    async def part_recv(self, timeout: float = None) -> bool:
+        """Wait for an event signalling the latest part segment."""
+        try:
+            async with async_timeout.timeout(timeout):
+                await self._part_event.wait()
+        except asyncio.TimeoutError:
+            return False
+        return True
+
+    def part_put(self) -> None:
+        """Set event signalling the latest part segment."""
+        self._part_event.set()
+        self._part_event.clear()
+
+    async def recv(self, timeout: float = None) -> Segment | None:
```

Comment:
How did it pass type checking? 🤔
```suggestion
    async def recv(self, timeout: float | None = None) -> Segment | None:
```

Reply:
Because it can be inferred: https://stackoverflow.com/questions/51055952/does-a-default-parameters-overwrite-type-hints-for-mypy

Reply:
That's nice, didn't know that.